src/qibolab/instruments/qm/controller.py

```diff
@@ -393,6 +393,14 @@ def register_acquisitions(
 
         return acquisitions
 
+    def _acquisition_id(self, probe_id: ChannelId) -> Optional[ChannelId]:
+        """Find id of acquisition channel corresponding to a given probe
+        channel."""
+        for id, channel in self.channels.items():
+            if isinstance(channel, AcquireChannel) and channel.probe == probe_id:
+                return id
+        return None
```

Comment:
In principle, this function is not very specific to QM, you could even lift it to the `channels` module, asking for both the arguments (i.e. `self.channels` and `probe_id`).

However, I truly don't know how much it will be required outside.

Reply:
The function was removed in bcf9c9df7d732bf9e3cbde73c68d5159c0ea257a following your next comment, so this one is probably not relevant anymore. If you agree with that commit I will merge. I tested that the routine still works.